server/mdm/android/service/androidmgmt/google_client.go

```diff
@@ -298,3 +325,56 @@ func getLastPart(ctx context.Context, name string) (string, error) {
 	}
 	return nameParts[len(nameParts)-1], nil
 }
+
+type appNotFoundError struct{}
+
+var _ fleet.NotFoundError = (*appNotFoundError)(nil)
+
+func (p appNotFoundError) Error() string {
+	return "Couldn’t add software. The application ID isn’t available in Play Store. Please find ID on the Play Store and try again."
+}
+
+func (p appNotFoundError) IsNotFound() bool {
+	return true
+}
+
+func (g *GoogleClient) EnterprisesApplications(ctx context.Context, enterpriseName, packageName string) (*androidmanagement.Application, error) {
+	if g == nil || g.mgmt == nil {
+		return nil, errors.New("android management service not initialized")
+	}
+
+	path := fmt.Sprintf("%s/applications/%s", enterpriseName, packageName)
+	app, err := g.mgmt.Enterprises.Applications.Get(path).Context(ctx).Do()
+	if err != nil {
+		var gapiErr *googleapi.Error
+		if errors.As(err, &gapiErr) {
+			if gapiErr.Code == http.StatusNotFound {
+				return nil, ctxerr.Wrap(ctx, appNotFoundError{})
+			}
+		}
+		return nil, fmt.Errorf("getting application %s: %w", packageName, err)
+	}
+	return app, nil
+}
+
+func (g *GoogleClient) EnterprisesPoliciesModifyPolicyApplications(ctx context.Context, policyName string, appPolicies []*androidmanagement.ApplicationPolicy) (*androidmanagement.Policy, error) {
+	var changes []*androidmanagement.ApplicationPolicyChange
+	for _, p := range appPolicies {
+		changes = append(changes, &androidmanagement.ApplicationPolicyChange{
+			Application: p,
+		})
+	}
+	req := androidmanagement.ModifyPolicyApplicationsRequest{
+		Changes: changes,
+	}
+	ret, err := g.mgmt.Enterprises.Policies.ModifyPolicyApplications(policyName, &req).Context(ctx).Do()
+	switch {
+	case googleapi.IsNotModified(err):
+		g.logger.Log("msg", "Android application policy not modified", "policy_name", policyName)
+		// TODO(JVE): does it make sense to return an error here?
```

Comment:
We do return that error in other GoogleClient methods so I think it'd be best to keep it consistent.